js/initcontext.go

```diff
@@ -165,22 +158,26 @@ func (i *InitContext) requireFile(name string) (goja.Value, error) {
 			}
 		}
 
-		pgm.exports = module.Get("exports")
 		i.programs[fileURL.String()] = pgm
 
 		// Run the program.
-		if _, err := i.runtime.RunProgram(pgm.pgm); err != nil {
+		f, err := i.runtime.RunProgram(pgm.pgm)
+		if err != nil {
+			delete(i.programs, fileURL.String())
 			return goja.Undefined(), err
 		}
-
-		pgm.exports = module.Get("exports")
+		if call, ok := goja.AssertFunction(f); ok {
+			if _, err = call(exports, pgm.module, exports); err != nil {
```

Comment:
I'm confused by this - the first argument here is the value of `this`, right? So shouldn't we call use the `pgm.module` for that instead of `exports`?

Reply:
Apperantly, not I first saw it in `https://github.com/dop251/goja_nodejs/blob/master/require/module.go#L102-L103` and while I can't find an actual documentation of nodejs that says this here is a [stackoverflow answer](https://stackoverflow.com/a/22771432/5427244) (or three). No idea why this (pun intended) is that way but :man_shrugging:

Reply:
I actually recommend looking at the [goja_nodejs' require](https://github.com/dop251/goja_nodejs/blob/master/require/module.go) as I think it is done very well (apart from not working with relative to the current module paths)

Reply:
:man_facepalming: so hard :face_with_head_bandage:... Oh well, ¯\_(ツ)_/¯...